src/components/virtual-scroll/virtual-scroll.ts

```diff
@@ -528,7 +528,13 @@ export class VirtualScroll implements DoCheck, AfterContentInit, OnDestroy {
   }
 
   private _updateDiffer() {
-    if (isBlank(this._differ) && isPresent(this._records)) {
```

Comment:
This was from the original code that initialized this._differ only when setting VirtualScroll property as a guard to not reinitialize the differ again. Now we need to initialize it twice. Once when setting VirtualScroll and once when setting TrackBy. If trackBy is not provided, then differ will be initialized with the default comparer. 
When DRYing the code in my pull request #11492 I accidently kept the original code that only accounted for differ being set once.